File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let start = 0;
        let end = nums.length - 1;
        while(start <= end) {
            let middle = Math.floor(start + (end - start) / 2);
            if(nums[middle] === target) return middle;

            if(nums[start] <= nums[middle]) {
                if(target >= nums[start] && target < nums[middle])
                    end = middle - 1;
                else
                    start = middle + 1;
            } else {
                if(target > nums[middle] && target <= nums[end])
                    start = middle + 1;
                else
                    end = middle - 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/merge-strings-alternately/submission-1.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1, l2 = len(word1), len(word2)
        start = 0
        new = ""
        while start < l1 and start < l2:
            new += word1[start] + word2[start]
            start += 1

        new += word1[start:]
        new += word2[start:]           

        return new